node/bft/src/primary.rs

```diff
@@ -1212,8 +1237,13 @@ impl<N: Network> Primary<N> {
         let self_ = self.clone();
         self.spawn(async move {
             while let Some((solution_id, solution, callback)) = rx_unconfirmed_solution.recv().await {
+                // Compute the checksum for the solution.
+                let Ok(checksum) = solution.to_checksum::<N>() else {
+                    error!("Failed to compute the checksum for the unconfirmed solution");
+                    continue;
+                };
                 // Compute the worker ID.
-                let Ok(worker_id) = assign_to_worker(solution_id, self_.num_workers()) else {
+                let Ok(worker_id) = assign_to_worker((solution_id, checksum), self_.num_workers()) else {
```

Comment:
A validator only stores and tries to verify unique `solution_id`s, it will ignore duplicates. So for assignment to workers, we can ignore the checksum. And because of https://github.com/AleoNet/snarkVM/pull/2520 there is no censorship risk.

Reply:
Workers are assigned based on `Into <TransmissionID>`, which is why we need the tuple there. 

An alternative, is to update `assign_to_worker` to intake a `ToBytes` object instead, which allows us to pass in just the solution ID or transaction ID.

Reply:
Thank you for clarifying. 

I see my above comment is wrong. We would significantly reduce (but not eliminate) risk of transmission censorship by just using the full `transmissionID` to determine whether a transmission was seen already. So I think we should rather use the full `transmissionID` as key in the `*_queue` and `seen_*`  caches in `node/consensus/src/lib.rs`, then we can just pass the full TransmissionID along so we don't have to recompute it here.